types/src/transaction/transaction_argument.rs

```diff
@@ -19,7 +19,7 @@ pub enum TransactionArgument {
     U64(u64),
     Address(AccountAddress),
     ByteArray(ByteArray),
-    String(String),
+    String(UserString),
```

Comment:
maybe if we drop UserString here and leave it to String we may have a simpler story (no LCS) and be as safe.
The entry point is effectively coming from outside and checking a String and creating a UserString seems a good model to me

Reply:
So, hmm, that's effectively what happens. We have two layers of structures: protobuf and domain-typed. The protobuf type is a plain old string while the domain type is a user string. A lot of validation happens in the protobuf -> domain type conversion, and any potential validation on user strings would happen there.

Reply:
I see, and fair point, maybe that is fine then, let me clarify where I was coming from with that comment. 
I liked the idea of UserString to stay internal to the VM and never leak out of it.
I suppose the question is whether TransactionArgument is internal to the VM or not and you see it already part of the VM (so to speak)